### PR TITLE
Fixes for iot tests

### PR DESCRIPTION
**Pull request by famartinrh**

### CODE REVIEW

**Comment by famartinrh** [comment]
@enmasse-ci run tests profile=shared-iot

File: systemtests/src/test/java/io/enmasse/systemtest/bases/IoTTestBase.java [hunk 1]
```diff
@@ -135,4 +146,29 @@ protected void createIoTProject(IoTProject project) throws Exception {
...
+                } else if(type == MessageType.TELEMETRY) {
+                    var response = adapterClient.sendTelemetry(json, any());
+                    return response.statusCode() == HTTP_ACCEPTED;
+                } else {
+                    return true;
+                }
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }, ofDuration(ofSeconds(60)));
```
Comment (dejanb) [inline systemtests/src/test/java/io/enmasse/systemtest/bases/IoTTestBase.java]: Maybe increase this timeout as well?

File: systemtests/src/test/java/io/enmasse/systemtest/bases/IoTTestBase.java [hunk 2]
```diff
@@ -135,4 +146,29 @@ protected void createIoTProject(IoTProject project) throws Exception {
...
+    protected void waitForFirstSuccessOnTelemetry(HttpAdapterClient adapterClient) throws Exception {
+        waitForFirstSuccess(adapterClient, MessageType.TELEMETRY);
+    }
+
+    protected void waitForFirstSuccess(HttpAdapterClient adapterClient, MessageType type) throws Exception {
+        JsonObject json = new JsonObject(Map.of("a", "b"));
+        TestUtils.waitUntilCondition("First successful "+type.name().toLowerCase()+" message", () -> {
+            try {
+                if(type == MessageType.EVENT) {
+                    var response = adapterClient.sendEvent(json, any());
```
Comment (dejanb) [inline systemtests/src/test/java/io/enmasse/systemtest/bases/IoTTestBase.java]: Can we add some logging here to see better what's going on when messages are not being processed properly?

File: systemtests/src/main/java/io/enmasse/systemtest/utils/WaitPhase.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright 2019, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.enmasse.systemtest.utils;
+
+public enum WaitPhase {
```
Comment (kornys) [inline systemtests/src/main/java/io/enmasse/systemtest/utils/WaitPhase.java]: this file should not be in utils package.

**Comment by famartinrh** [comment]
@enmasse-ci run tests profile=shared-iot

**Pull request merged**